card/mod/04_settings/lib/card/setting.rb

```diff
@@ -8,34 +8,36 @@ module Setting
     #     E = Card::Set::Self::Create (module extended with M)
     #     O = Card["*create"]         (object)
     attr_accessor :codename                                                # accessible in E
-    mattr_accessor :groups, :group_names, :user_specific                   # accessible in E and M 
-    def self.extended(host_class)     
+    mattr_accessor :groups, :group_names, :user_specific                   # accessible in E and M
+    def self.extended(host_class)
        host_class.mattr_accessor :restricted_to_type, :rule_type_editable  # accessible in E and O
     end
-    
-    @@groups = { :permission =>[], :look_and_feel=>[], :communication => [], :pointer=>[], :other =>[] }
+
     @@group_names = {
-      :permission    => "Permission",
-      :look_and_feel => "Look and Feel",
-      :communication => "Communication",
+      :permission    => "Permissions",
+      :event         => "Events",
+      :webpage       => "Webpage",
+      :templating    => "Templating",
+      :editing_cue   => "Editing cues",
       :pointer       => "Pointer",
-      :other         => "Other"  
+      :other         => "Other"
     }
+    @@groups = @@group_names.keys.inject({}) { |groups, key| groups[key] = []; groups }
```

Comment:
This is fine, but I've found each_with_object is usually simpler than inject.  Each with object only works as expected when you are mutating the object, which you are often when loading a hash.  So, @@group_names.keys.each_with_object({}) {|key, groups| groups[key] = [] }